src/radical/entk/pipeline.py

```diff
@@ -254,19 +256,32 @@ def from_dict(self, d):
 
         if 'name' in d:
             if d['name']:
+                invalid_symbols = punctuation.replace('.','')
+                if not isinstance(d['name'], str):
+                    raise ree.TypeError(expected_type=str,
+                                        actual_type=type(d['name']))
+
+                if any(symbol in d['name'] for symbol in invalid_symbols):
+                    raise ree.ValueError(obj=self._uid,
+                                        attribute='name',
+                                        actual_value=d['name'],
+                                        expected_value="Valid object names can " +
+                                        "contains letters, numbers and '.'. Any "
+                                        "other character is not allowed")
```

Comment:
Would it be worth to store these messages in a dedicated data structure? See that you repeated the same message twice, and now we have to fix the same error in both instances of the same message. More in general, should we template exceptions? Repeating `attribute` and other parameters all the time might be undesirable.

Reply:
I created a `constants` file to keep constants in general. I placed the message there. Some exceptions are already templated, but I am not a fan because it confuses me a bit. I can do it if you think it is important.

Reply:
This should only be done for messages we use a lot IMHO.

Reply:
This message is used ~8 times and each instance had the same typo. Happy for you to have the last word.

Reply:
@iparask, about the templating, happy to follow your preference as you are the main developer of EnTK. I would probably stress the need of consistency so that we do not end up with a mix of (non)templated exceptions?